java-checks/src/main/java/org/sonar/java/checks/SelfAssignementCheck.java

```diff
@@ -54,28 +69,77 @@ public void visitNode(Tree tree) {
     }
     AssignmentExpressionTree node = (AssignmentExpressionTree) tree;
     if (SyntacticEquivalence.areEquivalent(node.expression(), node.variable())) {
-      reportIssue(reportTree(node), ISSUE_MESSAGE);
+      QuickFixHelper.newIssue(context)
+        .forRule(this)
+        .onTree(reportTree(node))
+        .withMessage(ISSUE_MESSAGE)
+        .withQuickFix(() -> getQuickFix(node))
+        .report();
       updateWarnings(node);
     }
   }
 
+  private static JavaQuickFix getQuickFix(AssignmentExpressionTree tree) {
+    ClassTree classParent = (ClassTree) ExpressionUtils.getParentOfType(tree, Tree.Kind.CLASS);
+    MethodTree methodParent = (MethodTree) ExpressionUtils.getParentOfType(tree, Tree.Kind.METHOD, Tree.Kind.CONSTRUCTOR);
+    String name = getName(tree.variable());
+
+    if (methodParent != null && classParent != null) {
```

Comment:
This line is not fully covered in the tests and I am not sure we need to test if an assignment has a class parent in its hierarchy because a method can only exist in a class. Maybe we can simplify, and remove part of the condition?
```suggestion
    if (methodParent != null) {
```